extensions/ql-vscode/src/interface.ts

```diff
@@ -86,6 +86,22 @@ export function webviewUriToFileUri(webviewUri: string): Uri {
   return Uri.file(path);
 }
 
+function sortInterpretedResults(results: Sarif.Result[], sortState: InterpretedResultsSortState | undefined): void {
```

Comment:
No, it actually does get called in practice with an undefined `sortState` right now. That is the third state in the three-state cycle between `asc` and `desc` and 'don't sort'.

Reply:
To clarify, I'm suggesting moving this logic to only call `sortInterpretedResults` when `sortState` is not undefined, since an undefined `sortState` implies that there is no need to sort the results.

Reply:
Ah, I see. I think I still prefer this organization, where most of the code is 'in the option monad' and I'm just doing the `if` at the leaf here.

Reply:
I think it's slightly odd calling a function named "sort" when you don't intend to sort anything, but it's not a huge concern so I'll resolve the issue.